promotion/datastore.go

```diff
@@ -1696,6 +1849,44 @@ func (pg *Postgres) RunNextDrainRetryJob(ctx context.Context, worker DrainRetryW
 	}
 }
 
+// RunNextFetchRewardGrantsJob - runs fetch reward grant jobs
+func (pg *Postgres) RunNextFetchRewardGrantsJob(ctx context.Context, worker SwapRewardsWorker) error {
+
+	service, _ := InitService(ctx, pg, nil, nil)
+
+	grant, msg, err := worker.FetchRewardsGrants(ctx)
+	if err != nil {
+		return fmt.Errorf("fetch rewards grants job: failed to retrieve grant information: %w", err)
+	}
+
+	cohort := "control"
```

Comment:
@evq would it make sense to set the `cohort` to `swaps` for easier identification?